RRF-34: Admin screen

arianaantonio opened the pull request:
- Built account screen UI and add user modal
- Hooked up adding a new user
- Hooked up changing the current user
- Configured API with new user

arianaantonio committed:
RRF-34: Admin screen
- Built account screen UI and add user modal
- Hooked up adding a new user
- Hooked up changing the current user
- Configured API with new user

KingOfBrian commented on Demo/RCHAccountViewController.swift:
```diff
@@ -0,0 +1,130 @@
+            self.userIDArray = []
+            return
+        }
+        self.userIDArray = userIDArray
```
One style I like for these types of things is:

```
userIDArray = UserDefaults.standard.object(forKey: kRCHUserDefaultKeyUserIDs) as? [String] ?? []
```

KingOfBrian commented on Demo/RCHAccountViewController.swift:
```diff
@@ -0,0 +1,130 @@
+        if let userArray = UserDefaults.standard.object(forKey: kRCHUserDefaultKeyUserIDs) as? [String] {
+            mutableUserArray.addObjects(from: userArray)
+        }
+        mutableUserArray.add(newUser)
```
You should always try to avoid `NSMutable`'s if possible. Does this work?

```
var users = UserDefaults.standard.object(forKey: kRCHUserDefaultKeyUserIDs) as? [String] ?? []
users.append(newUser)
```

KingOfBrian commented on Demo/RCHAccountViewController.swift:
```diff
@@ -0,0 +1,130 @@
+    @IBOutlet var saveUserButton: UIButton!
+    @IBOutlet var userCellLabel: UILabel!
+    
+    var userIDArray: [String]?
```
#pp 
One approach I like is to model the state the view controller uses and have `didSet` manage the binding to the view layer.  This is scenario a bit different since the storage is in `UserDefaults`, and would use `set`, but for instance:

``` swift
var userIDs: [String] {
    get {
        return UserDefaults.standard.object(forKey: kRCHUserDefaultKeyUserIDs) as? [String] ?? []
    }
    set {
        UserDefaults.standard.set(newValue, forKey: kRCHUserDefaultKeyUserIDs)
        UserDefaults.standard.synchronize
        accountTableView.reloadData()
    }
}
var currentUser: String {
    // Same sort of relay into UserDefaults and reloadData()
}
```

Then else where you can things like:

``` swift
userIDs.append(newUser)
currentUser = newUser
```

And the view is updated as a side effect of managing state on the view controller. It will trigger `reloadData` twice, which isn't ideal, but for these scenarios should be pretty inconsequential, and there are some ways around that. Another application of this approach would be to have a `var displayModal: Bool` and in `didSet` update the hidden state and manage the first responder.

KingOfBrian commented:
#req Few Testing notes:
- On the Plus phones, the edges aren't stretching to the edge. I'm guessing the smaller phones might be clipping as well.
- I'd send some screenshots by the designers, a few notes:
    - The checkmark I think should go on the right side? I think there's an accessoryType you can use to get this pretty easily. The text left alignment jumping caught my eye
    - I think the row separator color is a bit off

arianaantonio committed:
RRF-34: Admin Screen
- Adjusted constraints for admin and client management screen
- Changed user id modal to UIAlertViewController
- Cleaned up code

arianaantonio committed:
RRF-34: Admin Screen
- Added a fatal error when trying to get current user
